services/src/main/java/org/keycloak/authentication/ClientAuthenticationFlow.java

```diff
@@ -96,6 +98,9 @@ public Response processFlow() {
                     }
 
                     logger.debugv("Client {0} authenticated by {1}", client.getClientId(), factory.getId());
+                    if (Boolean.parseBoolean(client.getAttribute(TEMP_ADMIN_ATTR_NAME))) {
+                        logger.warn(client.getClientId() + " is a temporary admin service account. To harden security, create a permanent account and delete the temporary one.");
+                    }
```

Comment:
@pedroigor Can you please confirm this is the right place to emit a warning during login?